kaitaistruct.py

```diff
@@ -327,7 +399,7 @@ def ensure_fixed_contents(self, expected):
         return actual
 
     @staticmethod
-    def bytes_strip_right(data, pad_byte):
+    def bytes_strip_right(data, pad_byte=b'\0'):
```

Comment:
This default value is wrong: the integer value of the byte is expected, not the "byte" itself. You would know this if you had inspected either to the function body or the generated Python code by KSC. If someone would rely on this default, it would keep returning the empty byte array regardless of the `data` (because the condition `data[new_len - 1] == pad_byte` would be always `False`, as you're comparing `int` to `bytes`, and `0 == b'0'` is `False` in both Python 2 and 3). 

Moreover, I also don't really see the need for adding the default values. As I said, KS runtime libraries are not meant to be _manual implementation helpers_.

And I can't recommend making that `pad_byte` parameter optional in this particular case anyway - it's the indispensable value for the method to work. Omitting it hides the intent. If I would encounter only `_io.bytes_strip_right(buf)`, I'd probably have to look in the method implementation _what on earth_ the method is stripping. Making it explicit like `_io.bytes_strip_right(buf, 0)` makes it clear.

You talk about _code readability_, but making parameters optional actually makes it worse in most cases.

Reply:
I see your point, thank you.